### PR TITLE
BUGZ-1247: Better handling of other-avatar verify-failed appearance

### PR DESCRIPTION
**Pull request by SimonWalton-HiFi**
Jira: https://highfidelity.atlassian.net/browse/BUGZ-1247
(Verify-failed avatar not appearing verify-failed part)

1. Add diagnostic for challenge ignored due to Wallet-salt not initialized.
2. Make verify-failed appearance (for other clients) more fundamental part of
AvatarData.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20931/HighFidelity-Beta-PR16069-2fafb68.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20931/HighFidelity-Beta-Interface-PR16069-2fafb68.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20931/HighFidelity-Beta-PR16069-2fafb68.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20931/HighFidelity-Beta-Interface-PR16069-2fafb68.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20931/HighFidelity-Beta-PR16069-2fafb68.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20931/HighFidelity-Quest-Beta-PR16069-2fafb68.apk)

File: libraries/avatars/src/AvatarHashMap.cpp [hunk 1]
```diff
@@ -330,10 +330,6 @@ void AvatarHashMap::processAvatarIdentityPacket(QSharedPointer<ReceivedMessage>
             bool displayNameChanged = false;
             // In this case, the "sendingNode" is the Avatar Mixer.
             avatar->processAvatarIdentity(avatarIdentityStream, identityChanged, displayNameChanged);
-            if (avatar->isCertifyFailed() && identityUUID != EMPTY) {
-                qCDebug(avatars) << "Avatar" << avatar->getSessionDisplayName() << "marked as VERIFY-FAILED";
```
Comment (howard-stearns) [inline libraries/avatars/src/AvatarHashMap.cpp]: do we still have something somewhere that records failure in the log?

Comment (SimonWalton-HiFi) [reply libraries/avatars/src/AvatarHashMap.cpp]: Not on the Interface side - I'll add that.

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20942/HighFidelity-Beta-PR16069-abad986.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20942/HighFidelity-Quest-Beta-PR16069-abad986.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20942/HighFidelity-Beta-PR16069-abad986.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20942/HighFidelity-Beta-Interface-PR16069-abad986.dmg

**Comment by SimonWalton-HiFi** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20943/HighFidelity-Beta-PR16069-abad986.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/20943/HighFidelity-Beta-Interface-PR16069-abad986.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20943/HighFidelity-Beta-PR16069-abad986.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/20943/HighFidelity-Beta-Interface-PR16069-abad986.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20943/HighFidelity-Beta-PR16069-abad986.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/20943/HighFidelity-Quest-Beta-PR16069-abad986.apk)

**Pull request merged**